### PR TITLE
v3.4.0 release notes

### PR DESCRIPTION
**Pull request by syurkevi**

### CODE REVIEW

File: docs/pages/release_notes.md [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
 ==============
+v3.4.0
+==============
+The source code with submodules can be downloaded directly from the following link:
+http://arrayfire.com/arrayfire_source/arrayfire-full-3.4.0.tar.bz2
+
+Improvements
+------------
+* New [interpolation methods](https://github.com/arrayfire/arrayfire/issues/1562) for resize, transform, approx1 and approx2
+* Cuda 8 and compute 6.x support
```
Comment (pavanky) [inline docs/pages/release_notes.md]: May be mention that the installer is still being built with 7.5 ?

File: docs/pages/release_notes.md [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+* Improved [JIT](https://github.com/arrayfire/arrayfire/issues/1472) evaluation heuristics
```
Comment (pavanky) [inline docs/pages/release_notes.md]: For CUDA and OpenCL

File: docs/pages/release_notes.md [hunk 3]
```diff
@@ -1,5 +1,63 @@
...
+* Correct time from [timeit()](https://github.com/arrayfire/arrayfire/pull/1414) function
+* Fix floating point numbers in [af::seq and missing size_of() types](https://github.com/arrayfire/arrayfire/pull/1404)
+* Explicitly extantiate [af::device() for void * ](https://github.com/arrayfire/arrayfire/issues/1503)
+
+Improvements
+------------
+* Cuda 8 and compute 6.x support, current installer still on 7.5
+* Improved [JIT](https://github.com/arrayfire/arrayfire/issues/1472) evaluation heuristics for CUDA and OpenCL
+* User controlled FFT plan caching
+* Cuda [speedups](https://github.com/arrayfire/arrayfire/pull/1411) for \ref wrap, \ref unwrap and \ref approx.
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: Cuda -> CUDA
`wrap()`, `unwrap()`, and one of:
`approx1()1 and `approx2()`, or
`[approx](\ref  approx_mat]

File: docs/pages/release_notes.md [hunk 4]
```diff
@@ -1,5 +1,63 @@
 Release Notes {#releasenotes}
 ==============
+v3.4.0
+==============
+The source code with submodules can be downloaded directly from the following link:
+http://arrayfire.com/arrayfire_source/arrayfire-full-3.4.0.tar.bz2
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: These lines are part of github release notes but not our documentation. You can remove these.

**Comment by shehzan10** [comment]
I have never really liked the way we handle github links. I suggest that at the end of each line we have something like...
```
this is my documentation line. [XYZ](https://github.com/arrayfire/arrayfire.....)
```
XYZ can be:
* Issue
* PR
* Code
* Commit
 * Or simply a generic "Link"

If we use `Link`, then we should only use link irrespective of what the link points to (issue/pr etc).
if we use Issue etc, then we should use Issue/PR/Code/Commit as relevant.

These are just my thought.

File: docs/pages/release_notes.md [hunk 5]
```diff
@@ -1,5 +1,63 @@
...
+
+Features
+----------
+* New [interpolation methods](https://github.com/arrayfire/arrayfire/issues/1562) for \ref af::resize(), \ref af::transform(), \ref af::approx1() and \ref af::approx2()
+* Support for complex mathematical functions
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: `[complex mathematical function](mathfunc__mat)`

File: docs/pages/release_notes.md [hunk 6]
```diff
@@ -1,5 +1,63 @@
...
+* New Forge graphics [integration](https://github.com/arrayfire/arrayfire/pull/1555)!
+    * Vector Field plotting functionality
+    * API updates
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: Removed GLEW and replaced with glbinding.
And links to both GLEW and glbinding.

File: docs/pages/release_notes.md [hunk 7]
```diff
@@ -1,5 +1,63 @@
...
+* \ref af::medfilt1(): [Median filter for 1-d signals](https://github.com/arrayfire/arrayfire/pull/1479)
+* af::RandomEngine(): New [random number generators](https://github.com/arrayfire/arrayfire/issues/868)//TODO
+    * Philox
+    * Threefry
+    * Mersenne Twister
+* \ref af::sparse(): [Sparse matrix support for all backends](https://github.com/arrayfire/arrayfire/issues/821)
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: [Sparse API](\ref sparse_func): [Sparse matrix support for all backends](https://github.com/arrayfire/arrayfire/issues/821)

File: docs/pages/release_notes.md [hunk 8]
```diff
@@ -1,5 +1,63 @@
...
+* \ref af::scan(): New [generalized scan](https://github.com/arrayfire/arrayfire/issues/388) functions
+* \ref af::moments(): New [image moments](https://github.com/arrayfire/arrayfire/pull/1453) functions
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: Use `[Image Moments](\ref moments_mat)`
Similar to sparse.
That way the entire group of functions can be linked.

File: docs/pages/release_notes.md [hunk 9]
```diff
@@ -1,5 +1,63 @@
...
+
+Bug Fixes
+--------------
+
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: Remove empty line between header and content.

File: docs/pages/release_notes.md [hunk 10]
```diff
@@ -1,5 +1,63 @@
...
+* Delegate [higher-dimension convolutions](https://github.com/arrayfire/arrayfire/pull/1445) to correct dimension
+* Indexing fixes with [c++11x](https://github.com/arrayfire/arrayfire/pull/1426), [af_indexers](https://github.com/arrayfire/arrayfire/pull/1426), and [empty arrays](https://github.com/arrayfire/arrayfire/issues/799)
+* Single element [median bugfix](https://github.com/arrayfire/arrayfire/pull/1423)
+* Correct time from [timeit()](https://github.com/arrayfire/arrayfire/pull/1414) function
+* Fix floating point numbers in [af::seq and missing size_of() types](https://github.com/arrayfire/arrayfire/pull/1404)
+* Explicitly extantiate [af::device() for void * ](https://github.com/arrayfire/arrayfire/issues/1503)
+
+Improvements
+------------
+* Cuda 8 and compute 6.x support, current installer still on 7.5
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: CUDA and Compute 6.x (Pascal)

**Comment by shehzan10** [comment]
Can you add glbinding information to [Copyright.md](https://github.com/arrayfire/arrayfire/blob/devel/COPYRIGHT.md).

File: docs/pages/release_notes.md [hunk 11]
```diff
@@ -1,5 +1,74 @@
...
+* Error message when [CUDA Compute Detection fails](https://github.com/arrayfire/arrayfire/issues/1535).
+* Several CMake build issues with Xcode generator fixed
+* Fix [multiple OpenCL definitions](https://github.com/arrayfire/arrayfire/issues/1429) at link-time
+* [Boost compute version update and lapacke detection fix](https://github.com/arrayfire/arrayfire/pull/1423)
+* Fix builds [with GCC 6.1.1](https://github.com/arrayfire/arrayfire/pull/1409) 
+
+Documentation
+-------------
+* Fixed grammar in license
+
```
Comment (shehzan10) [inline docs/pages/release_notes.md]: Add a "Known Issues" sections. Add:
```
Certain CUDA functions are known to be broken on Tegra K1. The following ArrayFire tests are currently failing:
* assign_cuda
* harris_cuda
* homography_cuda
* median_cuda
* orb_cudasort_cuda
* sort_by_key_cuda
* sort_index_cuda
```

**Comment by shehzan10** [comment]
Merged into #1588

**Pull request closed without merging**